WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -104,12 +104,11 @@ private extension PointOfSaleDashboardViewModel {
                 switch paymentState {
                 case .processingPayment,
                         .paymentError,
-                        .cardPaymentSuccessful:
-                    return true
-                case .idle,
-                        .acceptingCard,
+                        .cardPaymentSuccessful,
                         .validatingOrder,
                         .preparingReader:
```

Comment:
Disabling the button based on the current states on `orderState.isSyncing` doesn't seem to be enough, I was able to break the flow continuously by tapping the back button while the order is being validated/preparing. I've moved those two cases to always disable the back button

Reply:
It does make sense that we allow to come back only when the payment is waiting for the card or from a specific payment error view. 

It would be again great if you would merge with trunk, there's also a new `validatingOrderError` state from which I think we should also allow coming back from 👍